.github/workflows/tests.yml

```diff
@@ -28,12 +29,60 @@ jobs:
       run: |
         cd docs
         make html
+
+  ui-test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 40
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-22.04]
+        python: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Install node
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18.x'
+    - name: Install Python ${{ matrix.python }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Install Playwright
+      run: |
+          npm install -D @playwright/test
+          npx playwright install
+    - name: Install nbdime
+      run: |
+          pip install -e .[test]
+    - name: Install jupyterlab and dependencies
+      run: |
+          python -m pip install --upgrade pip wheel setuptools
+          python -m pip install jupyterlab~=4.0
```

Comment:
I would swap these two steps (first install latest pip to ensure resolution only then install nbdime).